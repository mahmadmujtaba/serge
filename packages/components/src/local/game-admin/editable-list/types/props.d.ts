export interface Item {
  name: string
  [property: string]: any
}

export default interface PropTypes {
  items: Array<Item>
  onClick?: (item: Item, e?: any) => void
  onChange?: (items: Array<Item>) => void
  onCreate?: () => void
  title?: string
}
