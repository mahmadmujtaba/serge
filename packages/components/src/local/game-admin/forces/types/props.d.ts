import { ForceData } from '@serge/custom-types'

export default interface PropTypes {
  forces: Array<ForceData>
  onChange: (forces: Array<ForceData>) => void
  onSave?: () => void
}

export { ForceData } from '@serge/custom-types'
export { Role } from '@serge/custom-types'
