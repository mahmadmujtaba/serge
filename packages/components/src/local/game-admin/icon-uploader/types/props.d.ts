export default interface PropTypes {
  onChange?: (src: string) => void
  background?: string
  icon?: string
  limit: number
}
