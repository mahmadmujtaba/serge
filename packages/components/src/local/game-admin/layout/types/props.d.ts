import { Wargame } from '@serge/custom-types';

export default interface PropTypes {
  /* half section title */
  type?: 'submit' | 'reset' | 'button'
  size?: 's' | 'm'
  onClick?: any
  disabled?: boolean
  wargame?: Wargame
  tabs?: Array<string>
}
