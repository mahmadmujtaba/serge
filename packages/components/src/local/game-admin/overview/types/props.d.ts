import { WargameOverview } from '@serge/custom-types'

export default interface PropTypes {
  overview: WargameOverview
  onChange: (nextOverview: WargameOverview) => void
  onSave?: () => void
}

export { WargameOverview } from '@serge/custom-types'
