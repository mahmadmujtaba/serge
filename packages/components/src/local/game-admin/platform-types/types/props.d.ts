import { PlatformType } from '@serge/custom-types'

export default interface PropTypes {
  platformType?: PlatformType
  onChange: (platformType: PlatformType) => void
  onSave?: () => void
}
