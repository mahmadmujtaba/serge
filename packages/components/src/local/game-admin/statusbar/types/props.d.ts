import { Wargame } from '@serge/custom-types';

export default interface PropTypes {
  /* half section title */
  href?: string
  onChange?: (e?: any) => void
  wargame: Wargame
}
