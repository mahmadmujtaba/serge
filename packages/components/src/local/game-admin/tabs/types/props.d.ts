export default interface PropTypes {
  /* half section title */
  onChange?: (selectedTab: string, e?: any) => void
  tabs: Array<string>
  activeTab?: string
}
