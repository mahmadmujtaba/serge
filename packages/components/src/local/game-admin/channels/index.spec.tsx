/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import Channels from './index'
import { ChannelData } from './types/props'
import { WargameExportedMock, MessageTemplatesMock } from '@serge/mocks'

const onChange = (channels: Array<ChannelData>): void => { console.log(channels) }

it('channels renders correctly', () => {
  const tree = renderer
    .create(<Channels
      channels={WargameExportedMock.data.channels.channels}
      forces={WargameExportedMock.data.forces.forces}
      onChange={onChange}
      messages={MessageTemplatesMock}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
