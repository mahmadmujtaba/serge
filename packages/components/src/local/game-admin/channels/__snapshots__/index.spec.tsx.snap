// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channels renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="left"
  >
    <div
      className="main"
    >
      <div
        className="section"
      >
        <button
          className="main"
          onClick={[Function]}
          type="button"
        >
          Add
        </button>
      </div>
      <div
        className="section"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink={false}
          >
            Search
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="search"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Search
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <ul>
        <li
          className=""
        >
          <div
            className="overlap"
            onClick={[Function]}
          />
          <div
            className="section"
          >
            <div>
              Channel 16
            </div>
            <span>
              <div
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-copy fa-w-14 "
                  data-icon="copy"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </div>
              <div
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash fa-w-14 "
                  data-icon="trash"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </div>
            </span>
          </div>
        </li>
        <li
          className=""
        >
          <div
            className="overlap"
            onClick={[Function]}
          />
          <div
            className="section"
          >
            <div>
              Blue Net
            </div>
            <span>
              <div
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-copy fa-w-14 "
                  data-icon="copy"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </div>
              <div
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash fa-w-14 "
                  data-icon="trash"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </div>
            </span>
          </div>
        </li>
        <li
          className=""
        >
          <div
            className="overlap"
            onClick={[Function]}
          />
          <div
            className="section"
          >
            <div>
              Mapping
            </div>
            <span>
              <div
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-copy fa-w-14 "
                  data-icon="copy"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </div>
              <div
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash fa-w-14 "
                  data-icon="trash"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </div>
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="right"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <input
            className="main"
            onChange={[Function]}
            placeholder="Channel Name"
            value="Channel 16"
          />
        </div>
        <div
          className="actions"
        >
          <button
            className="main"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col section table"
        >
          <div
            className="main"
          >
            <div
              className="placeholder"
            >
              Participants and messages
            </div>
            <div
              className="content"
            >
              <div
                className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <table
                  aria-label="simple table"
                  className="MuiTable-root"
                  role={null}
                >
                  <thead
                    className="MuiTableHead-root"
                    role={null}
                  >
                    <tr
                      className="MuiTableRow-root MuiTableRow-head"
                      role={null}
                    >
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-head"
                        scope="col"
                      >
                        Force
                      </th>
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                        scope="col"
                      >
                        Restrict access to specific roles
                      </th>
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                        scope="col"
                      >
                        Templates
                      </th>
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                        scope="col"
                      >
                        Actions
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="MuiTableBody-root"
                    role={null}
                  >
                    <tr>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          White
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          All roles
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Chat if empty
                        </div>
                      </td>
                      <td
                        align="right"
                        className="td"
                      >
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-pencil-alt fa-w-16 fa-sm "
                              data-icon="pencil-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Red
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          All roles
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Chat if empty
                        </div>
                      </td>
                      <td
                        align="right"
                        className="td"
                      >
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-pencil-alt fa-w-16 fa-sm "
                              data-icon="pencil-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Blue
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          All roles
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Chat if empty
                        </div>
                      </td>
                      <td
                        align="right"
                        className="td"
                      >
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-pencil-alt fa-w-16 fa-sm "
                              data-icon="pencil-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                      </td>
                    </tr>
                  </tbody>
                  <tfoot
                    className="MuiTableFooter-root"
                    role={null}
                  >
                    <tr>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          <div
                            className="MuiFormControl-root"
                          >
                            <div
                              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              onClick={[Function]}
                            >
                              <div
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-forces"
                                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                id="mui-component-select-forces"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onMouseDown={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                White
                              </div>
                              <input
                                aria-hidden={true}
                                className="MuiSelect-nativeInput"
                                name="forces"
                                onAnimationStart={[Function]}
                                onChange={[Function]}
                                required={false}
                                tabIndex={-1}
                                value="0"
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSelect-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          <div
                            className="MuiFormControl-root"
                          >
                            <div
                              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              onClick={[Function]}
                            >
                              <div
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-access"
                                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                id="mui-component-select-access"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onMouseDown={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                <div>
                                  All roles
                                </div>
                              </div>
                              <input
                                aria-hidden={true}
                                className="MuiSelect-nativeInput"
                                name="access"
                                onAnimationStart={[Function]}
                                onChange={[Function]}
                                required={false}
                                tabIndex={-1}
                                value=""
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSelect-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          <div
                            className="MuiFormControl-root"
                          >
                            <div
                              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              onClick={[Function]}
                            >
                              <div
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-templates"
                                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                id="mui-component-select-templates"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onMouseDown={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                <div>
                                  Chat if empty
                                </div>
                              </div>
                              <input
                                aria-hidden={true}
                                className="MuiSelect-nativeInput"
                                name="templates"
                                onAnimationStart={[Function]}
                                onChange={[Function]}
                                required={false}
                                tabIndex={-1}
                                value=""
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSelect-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </td>
                      <td
                        align="right"
                        className="td"
                      >
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-undo-alt fa-w-16 "
                              data-icon="undo-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M255.545 8c-66.269.119-126.438 26.233-170.86 68.685L48.971 40.971C33.851 25.851 8 36.559 8 57.941V192c0 13.255 10.745 24 24 24h134.059c21.382 0 32.09-25.851 16.971-40.971l-41.75-41.75c30.864-28.899 70.801-44.907 113.23-45.273 92.398-.798 170.283 73.977 169.484 169.442C423.236 348.009 349.816 424 256 424c-41.127 0-79.997-14.678-110.63-41.556-4.743-4.161-11.906-3.908-16.368.553L89.34 422.659c-4.872 4.872-4.631 12.815.482 17.433C133.798 479.813 192.074 504 256 504c136.966 0 247.999-111.033 248-247.998C504.001 119.193 392.354 7.755 255.545 8z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-check fa-w-16 fa-sm "
                              data-icon="check"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                      </td>
                    </tr>
                  </tfoot>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
